sgl-router/src/router.rs

```diff
@@ -403,25 +404,42 @@ impl Router {
     }
 
     fn get_text_from_request(&self, body: &Bytes, route: &str) -> String {
-        // convert body to json
-        let json = serde_json::from_slice::<serde_json::Value>(body).unwrap();
-
-        if route == "generate" {
-            // get the "text" field
-            let text = json.get("text").and_then(|t| t.as_str()).unwrap_or("");
-            return text.to_string();
-        } else if route == "v1/chat/completions" {
-            // get the messages field as raw text
-            if let Some(messages) = json.get("messages") {
-                // Convert messages back to a string, preserving all JSON formatting
-                return serde_json::to_string(messages).unwrap_or_default();
+        // Convert body to JSON
+        let json: Value = match serde_json::from_slice(body) {
+            Ok(j) => j,
+            Err(_) => {
+                warn!("Failed to parse JSON from request body.");
+                return String::new();
             }
-        } else if route == "v1/completions" {
-            let prompt = json.get("prompt").and_then(|t| t.as_str()).unwrap_or("");
-            return prompt.to_string();
-        }
+        };
 
-        return "".to_string();
+        match route {
+            "/generate" => {
+                // For /generate, always use the "text" field.
+                match json.get("text").and_then(Value::as_str) {
+                    Some(text) => text.to_string(),
+                    None => {
+                        warn!("No 'text' field found in request body for route /generate.");
```

Comment:
can we have a mode where we can just match token_ids? sglang engine itself doesn't have this limitation, right?

Reply:
Can you elaborate? This matches all text, formats as string, matches on per character level.

Reply:
Which part of the API spec are you talking about?